cypress/integration/context-selection/period/index.js

```diff
@@ -151,3 +174,28 @@ Then('the user should only see options of that type for "yearly"', () => {
             })
         })
 })
+
+Then('the user should only see options of that type for "quarterly"', () => {
+    cy.get('[data-test="period-selector-menu"] li').should('have.length', 4)
+})
+
+Then('the user should only see options of that type for "sixMonthly"', () => {
+    cy.get('[data-test="period-selector-menu"] li').should('have.length', 2)
+})
+
+Then(
+    'the user should only see options of that type for "financialApril"',
+    () => {
+        const currentYear = new Date().getFullYear()
+        const yearsToBeDisplayed = currentYear - 1970 + 2 // year of 0-timestamp, 2 openFuturePeriods for dataset
+        cy.get('[data-test="period-selector-menu"] li').should(
+            'have.length',
+            yearsToBeDisplayed
+        )
+    }
+)
+
+// count varies by year; for 2021: 53 weeks
+Then('the user should only see options of that type for "weekly"', () => {
+    cy.get('[data-test="period-selector-menu"] li').should('have.length', 53)
```

Comment:
The weekly period selection has been implemented with `periodId: 2021W52` as a constant to avoid this issue. I think that's probably okay rather than having a dynamic test that updates yearly? I think there's not a way to get around the yearly periodTypes being dynamic.